File: LuaProject/System/InitAbilitiesSystem.lua
local SystemBase = require("System.SystemBase")

---@class InitAbilitiesSystem : SystemBase, __ClassBase<InitAbilitiesSystem>
local cls = class("InitAbilitiesSystem", SystemBase)

function cls:Awake()
--#IF MAP_NAME_echoisles THEN
    -- 血DK
    require("Ability.DeathGrip")
    require("Ability.GorefiendsGrasp")
    require("Ability.DeathStrike")
    require("Ability.PlagueStrike")
    require("Ability.ArmyOfTheDead")
--#END

--#IF MAP_NAME_turtlerock THEN
    -- 邪DK
    require("Ability.FesteringWound")
    require("Ability.DeathCoil")
    require("Ability.Defile")
    require("Ability.Apocalypse")
    require("Ability.DarkTransformation")
    require("Ability.MonstrousBlow")
    require("Ability.ShamblingRush")
    require("Ability.PutridBulwark")
--#END

--#IF MAP_NAME_twistedmeadows THEN
    -- 默认 技能
    require("Ability.Evasion")
    require("Ability.MoonWellHeal")
    require("Ability.NativeRejuvenation")

    -- 武器战
    require("Ability.RageGenerator")
    require("Ability.DeepWounds")
    require("Ability.Overpower")
    require("Ability.Charge")
    require("Ability.MortalStrike")
    require("Ability.Condemn")
    require("Ability.BladeStorm")
--#END

--#IF MAP_NAME_moonglade THEN
    -- 唤魔师
    require("Ability.FireBreath")
    require("Ability.Disintegrate")
    require("Ability.SleepWalk")
    require("Ability.TimeWarp")
    require("Ability.MagmaBreath")
--#END

--#IF MAP_NAME_demo THEN
    -- 地穴领主
    require("Ability.PassiveDamageWithImpaleVisuals")

    -- 术士-克尔苏加德
    require("Ability.SoulSiphon")
    require("Ability.ShadowBolt")

    -- 牧师-希尔盖
    require("Ability.DarkHeal")
    require("Ability.DarkShield")
--#END
end

return cls

File: CSProject/Systems/InitAbilitiesSystem.cs
namespace Systems;

using LuaWrapper;
using SFLib;

[Lua(Class = "InitAbilitiesSystem")]
public class InitAbilitiesSystem : SystemBase
{
    public override void Awake()
    {
#if MAP_NAME_echoisles
        // 血DK
        LuaInterop.Require("Ability.DeathGrip");
        LuaInterop.Require("Ability.GorefiendsGrasp");
        LuaInterop.Require("Ability.DeathStrike");
        LuaInterop.Require("Ability.PlagueStrike");
        LuaInterop.Require("Ability.ArmyOfTheDead");
#endif

#if MAP_NAME_turtlerock
        // 邪DK
        LuaInterop.Require("Ability.FesteringWound");
        LuaInterop.Require("Ability.DeathCoil");
        LuaInterop.Require("Ability.Defile");
        LuaInterop.Require("Ability.Apocalypse");
        LuaInterop.Require("Ability.DarkTransformation");
        LuaInterop.Require("Ability.MonstrousBlow");
        LuaInterop.Require("Ability.ShamblingRush");
        LuaInterop.Require("Ability.PutridBulwark");
#endif

#if MAP_NAME_twistedmeadows
        // 默认 技能
        LuaInterop.Require("Ability.Evasion");
        LuaInterop.Require("Ability.MoonWellHeal");
        LuaInterop.Require("Ability.NativeRejuvenation");

        // 武器战
        LuaInterop.Require("Ability.RageGenerator");
        LuaInterop.Require("Ability.DeepWounds");
        LuaInterop.Require("Ability.Overpower");
        LuaInterop.Require("Ability.Charge");
        LuaInterop.Require("Ability.MortalStrike");
        LuaInterop.Require("Ability.Condemn");
        LuaInterop.Require("Ability.BladeStorm");
#endif

#if MAP_NAME_moonglade
        // 唤魔师
        LuaInterop.Require("Ability.FireBreath");
        LuaInterop.Require("Ability.Disintegrate");
        LuaInterop.Require("Ability.SleepWalk");
        LuaInterop.Require("Ability.TimeWarp");
        LuaInterop.Require("Ability.MagmaBreath");
#endif

#if MAP_NAME_demo
        // 地穴领主
        LuaInterop.Require("Ability.PassiveDamageWithImpaleVisuals");

        // 术士-克尔苏加德
        LuaInterop.Require("Ability.SoulSiphon");
        LuaInterop.Require("Ability.ShadowBolt");

        // 牧师-希尔盖
        LuaInterop.Require("Ability.DarkHeal");
        LuaInterop.Require("Ability.DarkShield");
#endif
    }
}
